adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,157 @@
+func (adapter *AlkimiAdapter) MakeRequests(request *openrtb2.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	reqCopy := *request
+	reqCopy.Imp = _updateImps(reqCopy)
+	encoded, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		reqBidder := _buildBidderRequest(adapter, encoded)
+		reqsBidder = append(reqsBidder, reqBidder)
+	}
+	return
+}
+
+func _updateImps(bidRequest openrtb2.BidRequest) []openrtb2.Imp {
+	updatedImps := make([]openrtb2.Imp, 0, len(bidRequest.Imp))
+	for _, imp := range bidRequest.Imp {
+		var impExtAlkimi openrtb_ext.ImpExtAlkimi
+		if err := json.Unmarshal(imp.Ext, &impExtAlkimi); err == nil {
+			var bidFloorPrice floors.Price
+			bidFloorPrice.FloorMinCur = imp.BidFloorCur
+			bidFloorPrice.FloorMin = imp.BidFloor
+
+			if len(bidFloorPrice.FloorMinCur) > 0 && bidFloorPrice.FloorMin > 0 {
+				imp.BidFloor = bidFloorPrice.FloorMin
+			} else {
+				imp.BidFloor = impExtAlkimi.BidFloor
+			}
+			imp.Instl = impExtAlkimi.Instl
+			imp.Exp = impExtAlkimi.Exp
+
+			extJson, err := json.Marshal(impExtAlkimi)
+			if err != nil {
+				continue
+			}
+			imp.Ext = extJson
+			updatedImps = append(updatedImps, imp)
+		}
+	}
+	return updatedImps
+}
+
+func _buildBidderRequest(adapter *AlkimiAdapter, encoded []byte) *adapters.RequestData {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    encoded,
+		Headers: headers,
+	}
+	return reqBidder
+}
+
+// MakeBids will parse the bids from the Alkimi server
+func (adapter *AlkimiAdapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if request == nil || response == nil || http.StatusNoContent == response.StatusCode {
+		return nil, nil
+	}
+
+	var bidResp openrtb2.BidResponse
+	err := json.Unmarshal(response.Body, &bidResp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	bidCount := len(bidResp.SeatBid)
+	if bidCount > 0 {
+		bidResponse := adapters.NewBidderResponseWithBidsCapacity(bidCount)
+		for _, seatBid := range bidResp.SeatBid {
+			for _, bid := range seatBid.Bid {
+				copyBid := bid
+				_resolveMacros(&copyBid)
+				impId := copyBid.ImpID
+				imp := request.Imp
+				bidType, _ := _getMediaTypeForImp(impId, imp)
+				bidderBid := &adapters.TypedBid{
+					Bid:     &copyBid,
+					BidType: bidType,
+					Seat:    "alkimi",
+				}
+				bidResponse.Bids = append(bidResponse.Bids, bidderBid)
+			}
+		}
+		return bidResponse, errs
+	}
+	return nil, nil
+}
+
+func _resolveMacros(bid *openrtb2.Bid) {
+	price := bid.Price
+	strPrice := fmt.Sprint(price)
+	bid.NURL = strings.ReplaceAll(bid.NURL, PRICE_MACRO, strPrice)
+	bid.AdM = strings.ReplaceAll(bid.AdM, PRICE_MACRO, strPrice)
```

Comment:
couple of points:
- should check if `bid.Price` is not 0 before replacing `PRICE_MACRO`
- should use ` strconv.FormatFloat` instead of fmt.Sprint
- may be error out if `bid.Price` is present and `bid.NURL` and `bid.AdM` has no `PRICE_MACRO`